esim/util.py

```diff
@@ -0,0 +1,252 @@
+
+		if t>0.5:
+			guess=True
+		else:
+			guess=False
+		label = test_data_label[i]
+		#print guess, label
+		if guess == True and label == False:
+			fp += 1.0
+		elif guess == False and label == True:
+			fn += 1.0
+		elif guess == True and label == True:
+			tp += 1.0
+		elif guess == False and label == False:
+			tn += 1.0
+		if label == guess:
+			counter += 1.0
+		#else:
+			#print label+'--'*20
+			# if guess:
+			# print "GOLD-" + str(label) + "\t" + "SYS-" + str(guess) + "\t" + sent1 + "\t" + sent2
+
+	try:
+		P = tp / (tp + fp)
+		R = tp / (tp + fn)
+		F = 2 * P * R / (P + R)
+	except:
+		P=0
+		R=0
+		F=0
+
+	#print "PRECISION: %s, RECALL: %s, F1: %s" % (P, R, F)
+	#print "ACCURACY: %s" % (counter/len(predict_result))
+	accuracy=counter/len(predict_result)
+
+	#print "# true pos:", tp
+	#print "# false pos:", fp
+	#print "# false neg:", fn
+	#print "# true neg:", tn
+	maxF1=0
+	P_maxF1=0
+	R_maxF1=0
+	probs = predict_result
+	sortedindex = sorted(range(len(probs)), key=probs.__getitem__)
+	sortedindex.reverse()
+
+	truepos=0
+	falsepos=0
+	for sortedi in sortedindex:
+		if test_data_label[sortedi]==True:
+			truepos+=1
+		elif test_data_label[sortedi]==False:
+			falsepos+=1
+		precision=0
+		if truepos+falsepos>0:
+			precision=truepos/(truepos+falsepos)
+
+		recall=truepos/(tp+fn)
+		f1=0
+		if precision+recall>0:
+			f1=2*precision*recall/(precision+recall)
+			if f1>maxF1:
+				#print probs[sortedi]
+				maxF1=f1
+				P_maxF1=precision
+				R_maxF1=recall
+	print("PRECISION: %s, RECALL: %s, max_F1: %s" % (P_maxF1, R_maxF1, maxF1))
+	return (accuracy, maxF1)
+
+def tokenize(text):
+	"""
+	Tokenize a piece of text using the Treebank tokenizer
+	:return: a list of strings
+	"""
+	return tokenizer.tokenize(text)
+	#return text.split()
+
+def shuffle_arrays(*arrays):
+	"""
+	Shuffle all given arrays with the same RNG state.
+	All shuffling is in-place, i.e., this function returns None.
+	"""
+	rng_state = np.random.get_state()
+	for array in arrays:
+		np.random.shuffle(array)
+		np.random.set_state(rng_state)
+
+def readSTSdata(dir):
+	#print(len(dict))
+	#print(dict['bmxs'])
+	lsents=[]
+	rsents=[]
+	labels=[]
+	for line in open(dir+'a.toks'):
+		line = line.decode('utf-8')
+		pieces=line.strip().split()
+		lsents.append(pieces)
+	for line in open(dir+'b.toks'):
+		line = line.decode('utf-8')
+		pieces = line.strip().split()
+		rsents.append(pieces)
+	for line in open(dir + 'sim.txt'):
+		sim = float(line.strip())
+		ceil = int(math.ceil(sim))
+		floor = int(math.floor(sim))
+		tmp = [0, 0, 0, 0, 0, 0]
+		if floor != ceil:
+			tmp[ceil] = sim - math.floor(sim)
+			tmp[floor] = math.ceil(sim) - sim
+		else:
+			tmp[floor] = 1
+		labels.append(tmp)
+	#data=(lsents,rsents,labels)
+	if not len(lsents)==len(rsents)==len(labels):
+		print('error!')
+		sys.exit()
+	clean_data = []
+	for i in range(len(lsents)):
+		clean_data.append((lsents[i], rsents[i], labels[i]))
+	return clean_data
+
+def readQuoradata(dir):
```

Comment:
Style is `read_quora_data`. Should we be putting _all_ the dataset utilities in a single `util.py`?

Reply:
Right. I should delete it since I never use read*() function.